projects/observability/src/shared/dashboard/widgets/top-n/data/top-n-data-source.model.ts

```diff
@@ -46,41 +33,62 @@ export class TopNDataSourceModel extends GraphQlDataSourceModel<TopNWidgetDataFe
 
   public getData(): Observable<TopNWidgetDataFetcher> {
     return of({
-      scope: this.entityType,
-      getData: (metricSpecification: MetricAggregationSpecification) => this.fetchDataWithMetric(metricSpecification)
+      scope: this.context,
+      getData: (metricSpecification: ExploreSpecification) => this.fetchDataWithMetric(metricSpecification)
     });
   }
 
-  private fetchDataWithMetric(metricSpecification: MetricAggregationSpecification): Observable<TopNWidgetValueData[]> {
-    return this.queryWithNextBatch<EntitiesGraphQlQueryHandlerService, EntitiesResponse>(filters => ({
-      requestType: ENTITIES_GQL_REQUEST,
-      entityType: this.entityType,
+  private readonly specBuilder: ExploreSpecificationBuilder = new ExploreSpecificationBuilder();
+
+  private readonly nameAttributeSpec: ExploreSpecification = this.specBuilder.exploreSpecificationForKey('name');
+  private readonly idAttributeSpec: ExploreSpecification = this.specBuilder.exploreSpecificationForKey('id');
+
+  private fetchDataWithMetric(metricSpecification: ExploreSpecification): Observable<TopNWidgetValueData[]> {
+    return this.queryWithNextBatch<ExploreGraphQlQueryHandlerService, GraphQlExploreResponse>(filters => ({
+      requestType: EXPLORE_GQL_REQUEST,
+      context: this.context,
       limit: this.resultLimit,
       timeRange: this.getTimeRangeOrThrow(),
-      properties: [this.attributeSpecification, metricSpecification],
+      selections: [this.nameAttributeSpec, this.idAttributeSpec, metricSpecification],
       filters: filters,
-      sort: {
-        direction: 'DESC',
-        key: metricSpecification
-      }
+      groupBy: {
+        keys: [this.nameAttributeSpec.name, this.idAttributeSpec.name]
+      },
+      orderBy: [
+        {
+          direction: 'DESC',
+          key: metricSpecification
+        }
+      ]
     })).pipe(
       map(response =>
         response.results.map(entity => ({
-          label: entity[this.attributeSpecification.resultAlias()] as string,
-          value: (entity[metricSpecification.resultAlias()] as MetricAggregation).value,
+          label: entity[this.nameAttributeSpec.resultAlias()].value as string,
```

Comment:
Similar to above comment, we're assuming this will be configured with an entity which isn't necessarily true in an explorer context. Need to either make it generic, or rename + scope down to entities.